src/Tribe/Rewrite.php

```diff
@@ -393,19 +426,22 @@ public function get_canonical_url( $url, $force = false ) {
 			$replaced = str_replace( array_keys( $replace ), $replace, $link_template );
 			// Remove trailing chars.
 			$path          = rtrim( $replaced, '?$' );
-			$canonical_url = trailingslashit( home_url( $path ) );
+			$resolved = trailingslashit( home_url( $path ) );
 
 			break;
 		}
 
-		$wp_canonical = redirect_canonical( $canonical_url, false );
-		$resolved     = empty( $wp_canonical ) ? $canonical_url : $wp_canonical;
+		if ( empty( $resolved ) ) {
+			$wp_canonical = redirect_canonical( $canonical_url, false );
+			$resolved     = empty( $wp_canonical ) ? $resolved : $wp_canonical;
+		}
+
 		if ( $canonical_url !== $resolved ) {
 			$resolved = trailingslashit( $resolved );
 		}
 
 		if ( count( $unmatched_vars ) ) {
-			$resolved = add_query_arg( $unmatched_vars, $canonical_url );
+			$resolved = add_query_arg( $unmatched_vars, $resolved );
 		}
 
```

Comment:
We should add a filter for the canonical URL.

Reply:
Right, thanks.